apps/vault/test/vault.js

```diff
@@ -96,19 +95,11 @@ contract('Vault app', (accounts) => {
       const testAccount = '0xbeef000000000000000000000000000000000000'
       const initialBalance = await getBalance(testAccount)
 
-      // Transfer with data param
-      await vault.transfer(ETH, testAccount, transferValue, NO_DATA)
+      // Transfer
+      await vault.transfer(ETH, testAccount, transferValue)
 
       assert.equal((await getBalance(testAccount)).toString(), initialBalance.add(transferValue).toString(), "should have sent eth")
       assert.equal((await getBalance(vault.address)).toString(), depositValue - transferValue, "should have remaining balance")
-
-      // Transfer without data param
-      /* Waiting for truffle to get overloading...
-      await vault.transfer(ETH, testAccount, transferValue)
-
-      assert.equal((await getBalance(testAccount)).toString(), initialBalance.add(transferValue * 2).toString(), "should have sent eth")
-      assert.equal((await getBalance(vault.address)).toString(), depositValue - transferValue * 2, "should have remaining balance")
-      */
```

Comment:
We could add a test with a mock that does too much in its fallback function (e.g. setting some state) to make sure we're only transferring 2.3k gas with `transfer()`.